File: 프로그래머스/lv1/양진혁/완주하지못한선수.py
def solution(participant, completion):
    dic = {}
    for p in participant:
        dic[p] = dic.get(p, 0) + 1
    for c in completion:
        if c in dic:
            dic[c] -= 1
        if dic[c] == 0:
            del dic[c]
    return list(dic.keys())[0]

File: 프로그래머스/lv1/김원래/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

unordered_map<string, int> uMap;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    for(int cnt = 0 ; cnt  < completion.size() ; ++cnt)
    {
        if(uMap.find(completion[cnt]) == uMap.end())
            uMap.insert({completion[cnt], 1});
        else
            uMap[completion[cnt]]++;
    }
    
    for(int cnt = 0 ; cnt < participant.size() ; ++cnt)
    {
        if(uMap.find(participant[cnt]) == uMap.end())
        {
            answer = participant[cnt];
            break;
        }
        
        if(uMap[participant[cnt]]-- == 0)
        {
            answer = participant[cnt];
            break;
        }
        
    }
    
    return answer;
}